armory/scenarios/poisoning_gtsrb_clbd.py

```diff
@@ -1,325 +1,61 @@
-"""
-Classifier evaluation within ARMORY
-
-Scenario Contributor: MITRE Corporation
-"""
-
+import copy
 import logging
-from typing import Optional
-import os
-import random
-from copy import deepcopy
 
 import numpy as np
 
-try:
-    from tensorflow import set_random_seed, ConfigProto, Session
-    from tensorflow.keras.backend import set_session
-    from tensorflow.keras.utils import to_categorical
-except ImportError:
-    from tensorflow.compat.v1 import (
-        set_random_seed,
-        ConfigProto,
-        Session,
-        disable_v2_behavior,
-    )
-    from tensorflow.compat.v1.keras.backend import set_session
-    from tensorflow.compat.v1.keras.utils import to_categorical
-
-    disable_v2_behavior()
-from tqdm import tqdm
-from PIL import ImageOps, Image
-
-from art.defences.trainer import AdversarialTrainerMadryPGD
-
-from armory.utils.config_loading import (
-    load_dataset,
-    load_model,
-    load,
-    load_fn,
-)
-from armory.utils import metrics
-from armory.scenarios.base import Scenario
+from armory.utils import config_loading
+from armory.scenarios.poison import DatasetPoisoner
+from armory.scenarios.poisoning_gtsrb_scenario import GTSRB
 
 logger = logging.getLogger(__name__)
 
 
-def poison_scenario_preprocessing(batch):
-    img_size = 48
-    img_out = []
-    quantization = 255.0
-    for im in batch:
-        img_eq = ImageOps.equalize(Image.fromarray(im))
-        width, height = img_eq.size
-        min_side = min(img_eq.size)
-        center = width // 2, height // 2
+class CleanDatasetPoisoner:
+    def __init__(self, attack):
+        self.attack = attack
 
-        left = center[0] - min_side // 2
-        top = center[1] - min_side // 2
-        right = center[0] + min_side // 2
-        bottom = center[1] + min_side // 2
+    def poison_dataset(self, x, y, return_index=False):
+        x_poison, y_poison = self.attack.poison(x, y)
```

Comment:
The main ART attack used in practice here (```PoisoningAttackCleanLabelBackdoor```) expects y in a one-hot format, and returns one-hot labels. It throws a numpy.AxisError when the y input is sparse. To avoid cluttering the scenario code, I guess we could modify the attack.

Reply:
I had removed some lines for `to_categorical` and back - I'll add them back in.